src/createHistoryMiddleware.js

```diff
@@ -0,0 +1,42 @@
+import ActionTypes from './ActionTypes';
+
+// This is a private action creator not intended to be called by users.
+function locationUpdated(location) {
+  return {
+    type: ActionTypes.LOCATION_UPDATED,
+    payload: location,
+  };
+}
+
+export default function createHistoryMiddleware(protocol) {
+  return function historyMiddleware({ dispatch }) {
+    const dispose = protocol.subscribe((location) => {
```

Comment:
bah, it tears down the listener here? sorry being dense. How does this interplay with TRANSITION or INIT for that matter?